crates/walrus-service/src/node/storage.rs

```diff
@@ -820,17 +821,18 @@ impl Storage {
     /// Test utility to get the shards that are live on the node.
     #[cfg(any(test, feature = "test-utils"))]
     pub async fn existing_shards_live(&self) -> Vec<ShardIndex> {
-        self.shards
-            .read()
-            .await
-            .values()
-            .filter_map(|shard_storage| {
-                shard_storage
-                    .status()
-                    .is_ok_and(|status| status.is_owned_by_node())
-                    .then_some(shard_storage.id())
-            })
-            .collect::<Vec<_>>()
+        let shards = self.shards.read().await;
+        let mut result = Vec::new();
+
+        for shard_storage in shards.values() {
+            if let Ok(status) = shard_storage.status().await
+                && status.is_owned_by_node()
+            {
+                result.push(shard_storage.id());
+            }
+        }
+
+        result
```

Comment:
Just in case you want to keep the iterator-like syntax:
```rust
        futures::future::join_all(shards.values().map(|shard_storage| async move {
            shard_storage
                .status()
                .await
                .is_ok_and(|status| status.is_owned_by_node())
                .then_some(shard_storage.id())
        }))
        .await
        .into_iter()
        .flatten()
        .collect()
```

Reply:
Applied. Thanks!